libs/vkd3d/command.c

```diff
@@ -4426,29 +4415,35 @@ static bool d3d12_command_list_create_framebuffer(struct d3d12_command_list *lis
     return true;
 }
 
-static bool d3d12_command_list_update_current_framebuffer(struct d3d12_command_list *list)
+static bool d3d12_command_list_update_rendering_info(struct d3d12_command_list *list)
 {
-    VkImageView views[D3D12_SIMULTANEOUS_RENDER_TARGET_COUNT + 2];
+    struct vkd3d_rendering_info *rendering_info = &list->rendering_info;
     struct d3d12_graphics_pipeline_state *graphics;
-    VkFramebuffer vk_framebuffer;
-    unsigned int view_count;
     uint32_t rtv_mask;
-    VkExtent3D extent;
     unsigned int i;
 
-    if (list->current_framebuffer != VK_NULL_HANDLE)
+    if (rendering_info->state_flags & VKD3D_RENDERING_CURRENT)
         return true;
 
     graphics = &list->state->graphics;
     rtv_mask = graphics->rtv_active_mask & list->rtv_nonnull_mask;
-    view_count = 0;
 
     /* The pipeline has fallback render passes / PSO in case we're
      * attempting to render to unbound RTV. */
-    while (rtv_mask)
+    for (i = 0; i < D3D12_SIMULTANEOUS_RENDER_TARGET_COUNT; i++)
     {
-        i = vkd3d_bitmask_iter32(&rtv_mask);
-        views[view_count++] = list->rtvs[i].view->vk_image_view;
+        VkRenderingAttachmentInfoKHR *attachment = &rendering_info->rtv[i];
```

Comment:
colorAttachmentCount should be updated here to reflect the colorAttachment count used in PSO I think.

Reply:
fixed, and changed the way we compute the pso rt count.